### ISSUE TITLE
Star Citizen and Anthem - game crash and freeze

### ISSUE THREAD

**Issue by ntropy83** [daniel-schuermann/mesa#44]
Just quickly gave the new driver a shot today in the morning on my two most heavy games.

Anthem does start the game window, exists immediantly tho and Star Citizen is running, I have the impression that the heavy shader compile stuttering is reduced a bit and my fps increased by 5. Yet after some minutes of playtime it crashes. I have dumbed the winelog:

https://pastebin.com/usgGeVQp

Keep up the good work :)
System: R7 2700X, 16 GB@3200 Mhz, Vega 64 and Ubuntu Budgie 18.04

**Comment by daniel-schuermann** [comment]
Ok, I think I found the issue already: would you be able to test this patch applied to the branch?
https://pastebin.com/dge2ZVwm

**Comment by daniel-schuermann** [comment]
The mentioned issue should be resolved now in the repository. Please open a new issue, if you encounter further problems. Thanks!

**Comment by ntropy83** [comment]
Hello Daniel,

I tried the new built but the problem isnt resolved yet. SC keeps crashing on compiling the shaders. I made a video with several attempts on ACO and one on LLVM to show the difference. 
I have tried a mesa-debug built aswell but the game did not start. So sadly I dont have any logs.

https://youtu.be/AkDU3LEsK24

Anthem dies too on launching with:
AL lib: (EE) ReleaseThreadCtx: Context 0x7f5ee8065460 current for thread being destroyed, possible leak!
AL lib: (EE) ReleaseThreadCtx: Context 0x7f5ee8065460 current for thread being destroyed, possible leak!
